logstash-core/lib/logstash/webserver.rb

```diff
@@ -0,0 +1,88 @@
+# encoding: utf-8
+require "puma"
+require 'puma/single'
+require 'puma/binder'
+require 'puma/configuration'
+require 'puma/commonlogger'
+
+module LogStash 
+  class WebServer
+
+    extend Forwardable
+
+    attr_reader :logger, :status, :config, :options, :cli_options, :runner, :binder, :events
+
+    def_delegator :@runner, :stats
+
+    def initialize(logger, options={})
+      @logger      = logger
+      http_port    = options[:http_port] || 9600
+      @options     = {}
+      @cli_options = options.merge({ :rackup => ::File.join(::File.dirname(__FILE__), "api", "init.ru"), 
+                                     :binds => ["tcp://0.0.0.0:#{http_port}"] })
```

Comment:
I think we should also support setting the bind host along with the port, many users may want to bind the web api to a local interface instead of all (which will expose it to the internet).